substrate/frame/support/procedural/src/pallet/expand/tasks.rs

```diff
@@ -0,0 +1,257 @@
+//! Contains logic for expanding task-related items.
+
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::pallet::{parse::tasks::*, Def};
+use derive_syn_parse::Parse;
+use inflector::Inflector;
+use proc_macro2::TokenStream as TokenStream2;
+use quote::{format_ident, quote, ToTokens};
+use syn::{parse_quote, spanned::Spanned, ItemEnum, ItemImpl};
+
+impl TaskEnumDef {
+	/// Since we optionally allow users to manually specify a `#[pallet::task_enum]`, in the
+	/// event they _don't_ specify one (which is actually the most common behavior) we have to
+	/// generate one based on the existing [`TasksDef`]. This method performs that generation.
+	pub fn generate(
+		tasks: &TasksDef,
+		type_decl_bounded_generics: TokenStream2,
+		type_use_generics: TokenStream2,
+	) -> Self {
+		let variants = match tasks.tasks_attr.is_some() {
+			true => tasks
+				.tasks
+				.iter()
+				.map(|task| {
+					let ident = &task.item.sig.ident;
+					let ident = format_ident!("{}", ident.to_string().to_class_case());
```

Comment:
I think this kills the span.

Reply:
it's getting quoted so it will get a new span anyway when this happens, this only matters for spans that we specifically pass to an `Error` AFAIK